terraform/deployments/elasticsearch-green/iam.tf

```diff
@@ -0,0 +1,63 @@
+resource "aws_iam_role" "manual_snapshot_role" {
+  name               = "${var.stackname}-elasticsearch6-manual-snapshot-role"
+  assume_role_policy = data.aws_iam_policy_document.es_can_assume_role.json
+}
+
+data "aws_iam_policy_document" "es_can_assume_role" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["es.amazonaws.com"]
+    }
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_policy" "manual_snapshot_bucket_policy" {
+  name   = "govuk-${var.govuk_environment}-${var.stackname}-elasticsearch6-manual-snapshot-bucket-policy"
+  policy = data.aws_iam_policy_document.manual_snapshot_bucket_policy.json
+}
+
+data "aws_iam_policy_document" "manual_snapshot_bucket_policy" {
+  statement {
+    actions   = ["s3:ListBucket"]
+    resources = var.elasticsearch6_manual_snapshot_bucket_arns
+  }
+  statement {
+    actions = [
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:DeleteObject",
+    ]
+    resources = formatlist("%s/*", var.elasticsearch6_manual_snapshot_bucket_arns)
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "manual_snapshot_role_policy" {
+  role       = aws_iam_role.manual_snapshot_role.name
+  policy_arn = aws_iam_policy.manual_snapshot_bucket_policy.arn
+
+}
+
+resource "aws_iam_policy" "can_configure_es_snapshots" {
+  name        = "govuk-${var.govuk_environment}-${var.stackname}-elasticsearch6-manual-snapshot-domain-configuration-policy"
```

Comment:
Double check this isn't longer than allowed

Reply:
An IAM Policy name [can be up to 128 characters in length](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html#reference_iam-quotas-entity-length). This should fit IMO.